machine/quanta/quanta_common_p2020/kernel/platform-quanta-common_p2020.patch

```diff
@@ -0,0 +1,415 @@
++{
++	struct mpic *mpic;
++	struct resource r;
++	struct device_node *np;
++
++	np = of_find_node_by_type(NULL, "open-pic");
++	if (np == NULL) {
++		printk(KERN_ERR "Could not find open-pic node\n");
++		return;
++	}
++
++	if (of_address_to_resource(np, 0, &r)) {
++		printk(KERN_ERR "Failed to map mpic register space\n");
++		of_node_put(np);
++		return;
++	}
++
++	mpic = mpic_alloc(np, r.start,
++		  MPIC_PRIMARY | MPIC_WANTS_RESET |
++		  MPIC_BIG_ENDIAN | MPIC_BROKEN_FRR_NIRQS |
++		  MPIC_SINGLE_DEST_CPU,
++		  0, 256, " OpenPIC  ");
++
++	BUG_ON(mpic == NULL);
++	of_node_put(np);
++
++	mpic_init(mpic);
++}
++
++/*
++ * Setup the architecture
++ */
++#ifdef CONFIG_SMP
++extern void __init mpc85xx_smp_init(void);
++#endif
++
++static void __init quanta_common_p2020_setup_arch(void)
++{
++	struct device_node *cpu;
++	const unsigned int *fp;
++#ifdef CONFIG_PCI
++	struct device_node *np;
++#endif
++
++	if (ppc_md.progress)
++		ppc_md.progress("quanta_common_p2020_setup_arch()", 0);
++
++        cpu = of_find_node_by_type(NULL, "cpu");
++        if (cpu != 0) {
++	     fp = of_get_property(cpu, "clock-frequency", NULL);
++	     if (fp != 0)
++		  loops_per_jiffy = *fp / HZ;
++	     else
++		  loops_per_jiffy = 500000000 / HZ;
++	     of_node_put(cpu);
++        }
++
++#ifdef CONFIG_PCI
++	for_each_node_by_type(np, "pci") {
++		if (of_device_is_compatible(np, "fsl,mpc8548-pcie"))
++			fsl_add_bridge(np, 0);
++	}
++#endif
++
++#ifdef CONFIG_SMP
++	mpc85xx_smp_init();
++#endif
++
++	printk(KERN_INFO "Quanta Common P2020 CPU board from Quanta Computer, Inc.\n");
++}
++
++
++
++static struct of_device_id __initdata quanta_common_p2020_ids[] = {
++	{ .type = "soc", },
++	{ .compatible = "soc", },
++	{ .compatible = "simple-bus", },
++	{ .compatible = "gianfar", },
++	{},
++};
++
++
++
++static int __init quanta_common_p2020_publish_devices(void)
++{
++	return of_platform_bus_probe(NULL, quanta_common_p2020_ids, NULL);
++}
++machine_device_initcall(quanta_common_p2020, quanta_common_p2020_publish_devices);
++
++
++
++static void quanta_common_p2020_show_cpuinfo(struct seq_file *m)
++{
++	uint pvid, svid, phid1;
++	uint memsize = total_memory;
++
++	pvid = mfspr(SPRN_PVR);
++	svid = mfspr(SPRN_SVR);
++
++	seq_printf(m, "Vendor\t\t: Freescale Semiconductor\n");
++	seq_printf(m, "PVR\t\t: 0x%x\n", pvid);
++	seq_printf(m, "SVR\t\t: 0x%x\n", svid);
++
++	/* Display cpu PLL setting */
++	phid1 = mfspr(SPRN_HID1);
++	seq_printf(m, "PLL setting\t: 0x%x\n", ((phid1 >> 24) & 0x3f));
++
++	/* Display the amount of memory */
++	seq_printf(m, "Memory\t\t: %d MB\n", memsize / (1024 * 1024));
++}
++
++
++/*
++ * Called very early, device-tree isn't unflattened
++ */
++static int __init quanta_common_p2020_probe(void)
++{
++	unsigned long root = of_get_flat_dt_root();
++	const char *compat_prefix = "quanta";
++	char *compat = of_get_flat_dt_prop(root, "compatible", NULL);
++
++	if(memcmp(compat, compat_prefix, strlen(compat_prefix)) == 0)
++		return 1;
```

Comment:
I presume you only match on the prefix "quanta" here to handle the rewriting of the platform name that comes after "quanta," ?